source/base/DateTime.ooc

```diff
@@ -14,22 +14,262 @@
 * You should have received a copy of the GNU Lesser General Public License
 * along with this software. If not, see <http://www.gnu.org/licenses/>.
 */
-TimeSpan: cover {
-	_ticks: Int64
-	ticks ::= this _ticks
-	init: func@ (=_ticks)
-	kean_base_timeSpan_getTicks: unmangled func -> Int64 { this _ticks }
-	new: unmangled(kean_base_timeSpan_new) static func ~API (ticks: UInt64) -> This { This new(ticks) }
+
+use ooc-base
+import os/Time
+
+DateTimeData: cover {
+	year: UInt32 { get set }
+	month: UInt8 { get set }
+	day: UInt8 { get set }
+	hour: UInt8 { get set }
+	minute: UInt8 { get set }
+	second: UInt8 { get set }
+	millisecond: Int { get set }
+  init: func@ ()
+	init: func@ ~fromDateTime (=year, =month, =day, =hour, =minute, =second, =millisecond)
+}
+
+extend Time {
+	dateTimeData: static func -> DateTimeData {
+		version(windows) {
+				st: SystemTime
+				GetLocalTime(st&)
+				return DateTimeData new(st wYear, st wMonth, st wDay, st wHour, st wMinute, st wSecond, st wMilliseconds)
+		}
+		version(!windows) {
+				tt := time(null)
+				tv : TimeVal
+				val := localtime(tt&)
+				gettimeofday(tv&, null)
+				return DateTimeData new(val@ tm_year+1900, val@ tm_mon+1, val@ tm_mday, val@ tm_hour, val@ tm_min, val@ tm_sec, tv tv_usec / 1000)
```

Comment:
I doubt we'll need it on Windows, but we will *need* it on Android. Millisecond precision isn't good enough.